qiskit/pulse/macros.py

```diff
@@ -0,0 +1,118 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Module for common pulse programming macros that construct pulse programs for common tasks."""
+
+from typing import Dict, List, Optional, Union
+
+from qiskit.pulse import channels
+from qiskit.pulse import commands
+from qiskit.pulse import exceptions
+from qiskit.pulse import instructions
+from qiskit.pulse import utils
```

Comment:
I know you felt like you had to do this. It really makes the rest of the file feel cluttered. In particular, `exception.XError` and `channel.XChannel` doesn't lend to readability and some of the line breaks that became necessary feel like they were done in awkward places. I suppose there's nothing we can do to make this better.

Reply:
At the end of the day, I had to do this and I prefer uniformity. However, as far as clutteredness I think its a matter of personal preference. I find it much easier to read the code and know where everything comes from and how they are related by their modules. I definitely can see where you are coming from and comes back to implicit vs explicit and the tradeoffs therein.